v-next/hardhat/test/internal/cli/main.ts

```diff
@@ -57,7 +62,199 @@ async function getTasksAndHreEnvironment(
+        const m = mock.method(console, "log", () => {});
+
+        const command = "npx hardhat --show-stack-traces example";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        assert.equal(m.mock.calls.length, 4);
+        assert.equal(m.mock.calls[2].arguments[0], "from a plugin");
+      });
+    });
+
+    describe("task with global flags and parameters", async function () {
+      useFixtureProject("cli/parsing/tasks-and-subtasks");
+
+      it("should run the task with global flags and parameters", async function () {
+        const command =
+          "npx hardhat --show-stack-traces task --param1 <value1> <value2> <value3>";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults();
+        assert.deepEqual(results, [true, true, true]);
+      });
+
+      it("should run the task with the default value", async function () {
+        const command = "npx hardhat task-default --show-stack-traces";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults();
+        assert.deepEqual(results, [true, false, false]);
+      });
+
+      it("should run the subtask with global flags and parameters", async function () {
+        const command =
+          "npx hardhat task subtask --param1 <value1> --show-stack-traces <value2> <value3>";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults();
+        assert.deepEqual(results, [true, true, true]);
+      });
+
+      it("should run the subtask with the default value", async function () {
+        const command =
+          "npx hardhat task-default --show-stack-traces subtask-default";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        const results = await getTasksAndSubtaskResults();
+        assert.deepEqual(results, [true, false, false]);
+      });
+    });
+
+    // TODO: as soon as the 'help task' is done, this test should be updated
+    describe("global help", function () {
+      useFixtureProject("cli/parsing/base-project");
+
+      it.todo("should print the global help", async function () {
+        const m = mock.method(console, "log", () => {});
+
+        const command = "npx hardhat";
+        const cliArguments = command.split(" ").slice(2);
+
+        await main(cliArguments);
+
+        assert.equal(m.mock.calls.length, 2);
+        assert.equal(m.mock.calls[1].arguments[0], "Global help");
+      });
+    });
+
+    // TODO: as soon as the 'help task' is done, this test should be updated
+    describe("subtask help", function () {
+      useFixtureProject("cli/parsing/subtask-help");
+
+      it.todo(
+        "should print an help message for the task's subtask",
```

Comment:
I think most of the time you can get away with either leaving out the todos (the follow task will have a testing requirement) or just putting in the string description.